ElrondNetwork/elrond-go
# Lite observer: do not start trie resolvers after bootstrap

## Reasoning behind the pull request
- lite observers won't be able to respond to trie requests
  
## Proposed changes
- create a new resolver that only makes the requests and do not register it on messenger as message processor for the topic

## Testing procedure
- standard system test
- standard system test, node started with `--operation-mode lite-observer`
- Local tests scenario passed:
   - start a node on each shard, on testnet, with `--operation-mode lite-observer`, wait to completely sync then wait for a epoch change
   - close the node, edit `config/p2p.toml` file and remove all entries from `InitialPeerList`
   - start the node again with the same operation-mode
   - search the logs for `listening on`, copy the path for each node and add them to `config/p2p.toml` file on a new node that will be started
   - start this new node with same opertaion mode
   - bootstrap should never finish on this one with the error `bootstrap process has failed: trie sync timeout`

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
